lms/djangoapps/badges/backends/badgr.py

```diff
@@ -166,12 +166,61 @@ def _create_assertion(self, badge_class, user, evidence_url):
         self._send_assertion_created_event(user, assertion)
         return assertion
 
-    @staticmethod
-    def _get_headers():
+    def _get_and_cache_oauth_tokens(self, refresh_token=None):
+        data = {
+            'username': settings.BADGR_USERNAME,
+            'password': settings.BADGR_PASSWORD,
+        }
+        if refresh_token:
+            data = {
+                'grant_type': 'refresh_token',
+                'refresh_token': refresh_token
+            }
+
+        oauth_url = "{}/o/token".format(settings.BADGR_BASE_URL)
+
+        response = requests.post(
+            oauth_url, data=data, timeout=settings.BADGR_TIMEOUT
+        )
+        response.raise_for_status()
+        try:
+            data = response.json()
+            # set the access token expiry 5 seconds early to make sure
```

Comment:
If I understand the code correctly, what the `expires_in` variable really does is set the the access token **cache** expiry; it's probably a good idea to say so in the comment.

Reply:
Hmm, so it's more the `access_token` expiry, I am not sure yet what to set the cache expiry (for now I set it to the same), we'd want to get the refresh token as long as possible because that is long-lived. So the cache expiry should outlive the `access_token` expiry.